src/core/store/modules/app/app.actions.ts

```diff
@@ -34,11 +36,16 @@ const clearUserAppData = createAsyncThunk<void, void, ThunkAPI>('app/clearUserAp
 /*                                 Fetch Data                                 */
 /* -------------------------------------------------------------------------- */
 
-const initApp = createAsyncThunk<void, void, ThunkAPI>('app/initApp', async (_arg, { dispatch, getState }) => {
+const initApp = createAsyncThunk<void, void, ThunkAPI>('app/initApp', async (_arg, { dispatch, getState, extra }) => {
+  const { CONTRACT_ADDRESSES } = extra.config;
   const { wallet, network } = getState();
```

Comment:
nit: Wondering if we should have some checks here for `wallet` and `network`.

Also, it's a bit nested, could we refactor this so we just return when we're done instead of having if/else?

Reply:
Since `wallet` and `network` have an initial state set by the reducer when the store gets created, dont think checks are necessary for wallet and network here at init.

Yes it does have some nested `if` conditions, but could not think of a better way to handle them :P

Reply:
All good, maybe we can refactor later, I have some ideas